packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -166,13 +197,74 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
             IUniswapV3Pool(wPowerPerpPool).fee(),
             _wPowerPerpAmount,
             amountToFlashswap,
-            uint8(FLASH_SOURCE.FLASH_W_MINT),
+            uint8(CALLBACK_SOURCE.FLASH_W_MINT),
             abi.encodePacked(_vaultId, amountToFlashswap, _collateralAmount, _wPowerPerpAmount)
         );
 
         emit FlashswapWMint(msg.sender, _vaultId, _wPowerPerpAmount, amountToFlashswap, _collateralAmount);
     }
 
+    /**
+     * @notice flash close position and buy long squeeth
+     * @dev this function
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmountToBurn amount of WPowerPerp to burn
+     * @param _wPowerPerpAmountToBuy amount of WPowerPerp to buy
+     * @param _collateralToWithdraw amount of collateral to withdraw from vault
+     * @param _maxToPay max amount of collateral to pay for WPowerPerp token
+     */
+    function flashswapWBurnBuyLong(
+        uint256 _vaultId,
+        uint256 _wPowerPerpAmountToBurn,
+        uint256 _wPowerPerpAmountToBuy,
+        uint256 _collateralToWithdraw,
+        uint256 _maxToPay
+    ) external payable {
+        require(_maxToPay <= _collateralToWithdraw.add(msg.value), "Not enough collateral");
+
+        _exactOutFlashSwap(
+            weth,
+            wPowerPerp,
+            IUniswapV3Pool(wPowerPerpPool).fee(),
+            _wPowerPerpAmountToBurn.add(_wPowerPerpAmountToBuy),
+            _maxToPay,
+            uint8(CALLBACK_SOURCE.FLASH_W_BURN),
+            abi.encodePacked(_vaultId, _wPowerPerpAmountToBurn, _wPowerPerpAmountToBuy, _collateralToWithdraw)
+        );
+
+        emit FlashWBurn(msg.sender, _vaultId, _wPowerPerpAmountToBurn, _collateralToWithdraw, _wPowerPerpAmountToBuy);
+    }
+
+    /**
+     * @notice sell long wPowerPerp and flashswap mint short position
+     * @dev flahswap amount = collateral amount - msg.value - ETH from selling long wPowerPerp
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmountToMint wPowerPerp amount to mint
+     * @param _collateralAmount collateral amount to use for minting
+     * @param _wPowerPerpAmountToSell long wPowerPerp amount to sell
+     * @param _minToReceive min ETH amount to receive for selling long wPowerPerp amount
```

Comment:
this is min to receive for sum of _wPowerPerpAmountToMint and _wPowerPerpAmountToSell